rust-wasm-id-allocator/distributed-id-allocator/src/compressor/tables/uuid_space.rs

```diff
@@ -59,6 +59,38 @@ impl UuidSpace {
         }
     }
 
+    pub fn range_can_collide(
+        &self,
+        originator: SessionId,
+        sessions: &Sessions,
+        range_base: StableId,
+        range_max: StableId,
+    ) -> bool {
+        let mut range = self
+            .uuid_to_cluster
+            .range((
+                Bound::Excluded(StableId::null()),
+                Bound::Included(range_max),
+            ))
+            .rev();
+        match range.next() {
+            None => return false,
+            Some((_, &cluster_ref)) => {
+                let cluster_match = sessions.deref_cluster(cluster_ref);
+                let result_session_id = sessions
+                    .deref_session_space(cluster_match.session_creator)
+                    .session_id();
+                let cluster_max_stable =
+                    result_session_id + cluster_match.base_local_id + cluster_match.capacity;
+                if originator != result_session_id && range_base <= cluster_max_stable {
```

Comment:
Note that this implementation specifically does not check for collision with a session's own clusters. Non-collision with a session's own clusters is mathematically impossible outside of an implementation bug, and is not related to the user action which this check is intended to address (creating a new session with a close incrementation of an existing SessionId). 

Allowing for checking collision with a session's own clusters adds appreciable complexity.